source/thisTarget.ts

```diff
@@ -7,12 +7,39 @@ import { messenger } from "./sender.js";
 import { registerMethods } from "./receiver.js";
 import { AnyTarget, MessengerMeta, Sender } from "./types.js";
 import { debug } from "./shared.js";
+import { Entries } from "type-fest";
 
 // Soft warning: Race conditions are possible.
 // This CANNOT be awaited because waiting for it means "I will handle the message."
 // If a message is received before this is ready, it will just have to be ignored.
 let thisTarget: AnyTarget | undefined;
 
+function compareTargets(to: AnyTarget, thisTarget: AnyTarget): boolean {
```

Comment:
I looked into using URLPattern to compare the URLs, but unfortunately that API does not care for search parameters because it treats them as a plain string so only exact matches are possible. 

This function instead will match the target `a.html?id=1` even if the current page is `a.html?something=else&id=1`